tracer/src/Datadog.Trace/Tagging/TraceSourcesUtils.cs

```diff
@@ -0,0 +1,84 @@
+// <copyright file="TraceSourcesUtils.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.Tagging
+{
+    /// <summary>
+    /// Utility funtions to handle TraceSources values in the '_dd.p.ts' tag
+    /// </summary>
+    internal static class TraceSourcesUtils
```

Comment:
Following .NET naming conventions, we should probably call this `TraceTagCollectionExtensions`.

Reply:
For perf reasons, is would help to add a `TraceContext.TraceSources` instead of using `TraceContext.Tags` directly. But that means we also need to
- inject the tag during span serialization
- inject the tag into downstream propagation headers
- extract the tag from upstream propagation headers

Not sure if it's worth the work, and even if we do it, it should be a separate PR from this one.

Reply:
Yes, I considered that but thought it was going to bee too much hassle. But noted for a different PR.